tests/integration-tests/tests/api/test_api_client.py

```diff
@@ -136,3 +143,88 @@ def _test_delete_cluster(client, cluster, region):
+
+
+@pytest.mark.usefixtures("instance")
+def test_custom_image(request, region, os, pcluster_config_reader, api_client):
+    base_ami = retrieve_latest_ami(region, os)
+
+    config_file = pcluster_config_reader(config_file="image.config.yaml", parent_image=base_ami)
+    with open(config_file) as config_file:
+        config = config_file.read()
+
+    stack_name = generate_stack_name("integ-tests-build-image", request.config.getoption("stackname_suffix"))
+    client = image_operations_api.ImageOperationsApi(api_client)
+
+    with soft_assertions():
+        resp = _test_build_image(client, config, region, stack_name)
+        LOGGER.info("Build image response: %s", resp)
+        image = resp["image"]
+        image_id = image["image_id"]
+        assert_that(image_id).is_equal_to(stack_name)
+
+        _test_describe_image(client, image, region)
+        _test_list_images(client, image, region, "PENDING")
+
+        # CFN stack is deleted as soon as image is available
+        _cloudformation_wait(region, stack_name, "stack_delete_complete")
+
+        _test_describe_image(client, image, region)
+        _test_list_images(client, image, region, "AVAILABLE")
+
+        _delete_image(client, image, region)
+
+
+def _test_build_image(client, image_config, region, image_id):
+    image_config_data = base64.b64encode(image_config.encode("utf-8")).decode("utf-8")
+    body = BuildImageRequestContent(image_config_data, image_id, region=region)
+    return client.build_image(body)
+
+
+def _test_describe_image(client, image, region):
+    image_id = image["image_id"]
+    response = client.describe_image(image_id, region=region)
+    LOGGER.info("Describe image response: %s", response)
+    assert_that(response.image_id).is_equal_to(image_id)
+
+
+def _test_list_images(client, image, region, status):
+    image_id = image["image_id"]
+
+    response = client.list_images(image_status=ImageStatusFilteringOption(status), region=region)
+    target_image = _get_image(response["items"], image_id)
+    next_token = response.get("next_token", None)
+
+    while next_token and not target_image:
+        response = client.list_clusters(
+            image_status=ImageStatusFilteringOption(status), region=region, next_token=next_token
+        )
+        target_image = _get_cluster(response["items"], image_id)
+        next_token = response.get("next_token", None)
+
+    LOGGER.info("ListImage contains target image: %s", target_image)
+
+    assert_that(target_image).is_not_none()
+
+
+def _get_image(images, image_id):
+    for image in images:
+        if image["image_id"] == image_id:
+            return image
+    return None
+
+
+def _delete_image(client, image, region):
+    image_id = image["image_id"]
+    client.delete_image(image_id, region=region)
+
+    # Deletion of available image is synchronous, so we do not have to wait as in the cluster case
+    response = client.list_images(image_status="AVAILABLE", region=region)
```

Comment:
what about a describe_image which is faster and expect the not found exception?

Reply:
Sure